haddock-library/src/Documentation/Haddock/Parser/Monad.hs

```diff
@@ -51,120 +29,86 @@ newtype ParserState = ParserState {
-    else fail "satsifyUnicode"
-
-anyChar :: Parser Char
-anyChar = lift Attoparsec.anyChar
-
-notChar :: Char -> Parser Char
-notChar = lift . Attoparsec.notChar
+char =  Parsec.char
+
+many' :: Parser a -> Parser [a]
+many' = Parsec.manyAccum (\x xs -> x `seq` x : xs)
 
 satisfy :: (Char -> Bool) -> Parser Char
-satisfy = lift . Attoparsec.satisfy
+satisfy = Parsec.satisfy
 
 peekChar :: Parser (Maybe Char)
-peekChar = lift Attoparsec.peekChar
+peekChar = Parsec.optionMaybe . Parsec.try . Parsec.lookAhead $ Parsec.anyChar
 
 peekChar' :: Parser Char
-peekChar' = lift Attoparsec.peekChar'
+peekChar' = Parsec.lookAhead $ Parsec.anyChar 
 
 digit :: Parser Char
-digit = lift Attoparsec.digit
-
-letter_iso8859_15 :: Parser Char
-letter_iso8859_15 = lift Attoparsec.letter_iso8859_15
-
-letter_ascii :: Parser Char
-letter_ascii = lift Attoparsec.letter_ascii
+digit = Parsec.digit
 
 space :: Parser Char
-space = lift Attoparsec.space
-
-string :: ByteString -> Parser ByteString
-string = lift . Attoparsec.string
+space = Parsec.space
 
-stringCI :: ByteString -> Parser ByteString
-stringCI = lift . Attoparsec.stringCI
+string :: Text -> Parser Text
+string = fmap T.pack . Parsec.string . T.unpack
 
 skipSpace :: Parser ()
-skipSpace = lift Attoparsec.skipSpace
+skipSpace = Parsec.skipMany Parsec.space
 
 skipWhile :: (Char -> Bool) -> Parser ()
-skipWhile = lift . Attoparsec.skipWhile
-
-take :: Int -> Parser ByteString
-take = lift . Attoparsec.take
+skipWhile = Parsec.skipMany . Parsec.satisfy
 
-scan :: s -> (s -> Char -> Maybe s) -> Parser ByteString
-scan s = lift . Attoparsec.scan s
+take :: Int -> Parser Text
+take = fmap T.pack . go
+  where go !n | n <= 0 = pure ""
+              | otherwise = liftA2 (:) Parsec.anyChar (go (n - 1)) <|> pure ""
 
-takeWhile :: (Char -> Bool) -> Parser ByteString
-takeWhile = lift . Attoparsec.takeWhile
+scan :: s -> (s -> Char -> Maybe s) -> Parser Text 
+scan s f = fmap T.pack (go s)
+  where go s1 = do { cOpt <- peekChar
+                   ; case cOpt >>= f s1 of
+                       Nothing -> pure ""
+                       Just s2 -> liftA2 (:) Parsec.anyChar (go s2)
+                   }
 
-takeWhile1 :: (Char -> Bool) -> Parser ByteString
-takeWhile1 = lift . Attoparsec.takeWhile1
+takeWhile :: (Char -> Bool) -> Parser Text
+takeWhile = fmap T.pack . Parsec.many . Parsec.satisfy
 
-takeTill :: (Char -> Bool) -> Parser ByteString
-takeTill = lift . Attoparsec.takeTill
-
-takeByteString :: Parser ByteString
-takeByteString = lift Attoparsec.takeByteString
-
-takeLazyByteString :: Parser LB.ByteString
-takeLazyByteString = lift Attoparsec.takeLazyByteString
+takeWhile1 :: (Char -> Bool) -> Parser Text
+takeWhile1 =  fmap T.pack . Parsec.many1 . Parsec.satisfy
 
 endOfLine :: Parser ()
-endOfLine = lift Attoparsec.endOfLine
+endOfLine = void Parsec.endOfLine
 
 decimal :: Integral a => Parser a
-decimal = lift Attoparsec.decimal
+decimal = foldl' step 0 `fmap` Parsec.many1 (satisfy isDigit)
```

Comment:
https://hackage.haskell.org/package/parsec-3.1.13.0/docs/Text-Parsec-Char.html#v:digit ; same for `hexadecimal`